### PR TITLE
Change MultiHateClassification to use dataset_transform

### PR DESCRIPTION
**Pull request by dokato**
Change MultiHateClassification to use dataset_transform istead of overloading load_dataset.

As discussed with @KennethEnevoldsen in https://github.com/embeddings-benchmark/mteb/pull/545

### CODE REVIEW

**Review by imenelydiaker — COMMENTED** [review]
Can you please the 2 porposed evaluations on this task?

**Review by KennethEnevoldsen — COMMENTED** [review]
Looks good! It would be wonderful to double check that it does influence the scores (as @imenelydiaker suggested) but otherwise feel free to add points (2 for the fix seems reasonable to me)

**Comment by dokato** [comment]
Thanks both! And yes, indeed I did run the evaluation, the results come out all the same.

File: docs/mmteb/points/558.jsonl [hunk 1]
```diff
@@ -0,0 +1 @@
+{"GitHub": "dokato", "Bug fix": 2}
```
Comment (KennethEnevoldsen) [inline docs/mmteb/points/558.jsonl:1]:
```suggestion
{"GitHub": "dokato", "Bug fixes": 2}
{"GitHub": "imenelydiaker", "Review PR": 2}
{"GitHub": "KennethEnevoldsen", "Review PR": 2}

```

**Review by KennethEnevoldsen — APPROVED** [review]

**Comment by KennethEnevoldsen** [comment]
set to auto merge

**Pull request merged**